### DISCUSSION TITLE
for Ref , Ubuntu make fails , but fprime-util build  works

### DISCUSSION THREAD

**Discussion by knapsea** [nasa/fprime#871] [Q&A] [answered]
garden@Bunna:~/fprime/Ref$ make clean
garden@Bunna:~/fprime/Ref$ pwd
/home/fprime/Ref
garden@Bunna:~/fprime/Ref$ make 
....

arsing Interface CmdResponse
Parsing Interface Tlm
Parsing Serializable SignalPair
Parsing Serializable SignalInfo
[ 30%] Generating SignalTypeEnumAc.cpp, SignalTypeEnumAc.hpp
Parsing Enum SignalType
Completed generating files for /home/garden/fprime/Ref/SignalGen/SignalTypeEnumAi.xml Enum XML....
[ 30%] Generating SignalSetArrayAc.cpp, SignalSetArrayAc.hpp
Parsing Array SignalSet
Completed generating files for /home/garden/fprime/Ref/SignalGen/SignalSetArrayAi.xml Array XML...
[ 30%] Generating SignalPairSerializableAc.cpp, SignalPairSerializableAc.hpp
Parsing Serializable SignalPair
Generating code filename: SignalPairSerializableAc.cpp, using default XML filename prefix...
Generating code filename: SignalPairSerializableAc.hpp, using default XML filename prefix...
[ 30%] Generating SignalPairSetArrayAc.cpp, SignalPairSetArrayAc.hpp
Parsing Array SignalPairSet
Completed generating files for /home/garden/fprime/Ref/SignalGen/SignalPairSetArrayAi.xml Array XML...
[ 30%] Generating SignalInfoSerializableAc.cpp, SignalInfoSerializableAc.hpp
Parsing Serializable SignalInfo
Generating code filename: SignalInfoSerializableAc.cpp, using default XML filename prefix...
Generating code filename: SignalInfoSerializableAc.hpp, using default XML filename prefix...
[ 30%] Building CXX object SignalGen/CMakeFiles/Ref_SignalGen.dir/SignalGen.cpp.o
In file included from /home/garden/fprime/Ref/SignalGen/SignalGen.cpp:14:
/home/garden/fprime/Ref/SignalGen/SignalGen.hpp:20:10: fatal error: Ref/SignalGen/SignalGenComponentAc.hpp: No such file or directory
   20 | #include <Ref/SignalGen/SignalGenComponentAc.hpp>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
make[2]: *** [SignalGen/CMakeFiles/Ref_SignalGen.dir/build.make:140: SignalGen/CMakeFiles/Ref_SignalGen.dir/SignalGen.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:10002: SignalGen/CMakeFiles/Ref_SignalGen.dir/all] Error 2
make: *** [Makefile:84: all] Error 2

garden@Bunna:~/fprime/Ref$ fprime-util build --jobs "$(nproc || printf '%s\n' 1)"
garden@Bunna:~/fprime/Ref$ fprime-gds -g html -r /home/kmajit/fprime/Ref/build-artifacts

**Comment by LeStarch** [comment, marked as answer]
If you are going to "make" directly, you need to build within a cmake build-cache.

```
fprime-util generate # Creates a cmake-cache at build-fprime-automatic-native
fprime-util build # cd  build-fprime-automatic-native; make
```

You could try this by hand:

```
mkdir Ref/build-cache
cd Ref/build-cache
cmake ..
make
```
However, this gets complicated quickly if/when you used `settings.ini` or more modern project organization.  That is why we recommend using fprime-util.
